xdsl/transforms/mlir_opt.py

```diff
@@ -0,0 +1,50 @@
+import shutil
+import subprocess
+from dataclasses import dataclass, field
+from io import StringIO
+
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.ir.core import MLContext
+from xdsl.parser.core import Parser
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import PatternRewriter
+from xdsl.printer import Printer
+
+
+@dataclass
+class MLIROptPass(ModulePass):
+    """
+    A pass for calling the `mlir-opt` tool with specified parameters.
+    """
+
+    name = "mlir-opt"
+
+    arguments: list[str] = field(default_factory=list)
+
+    def apply(self, ctx: MLContext, op: ModuleOp) -> None:
+        if not shutil.which("mlir-opt"):
+            raise ValueError("mlir-opt is not available")
```

Comment:
Does this work platform-independently?

Reply:
I have no idea :) what platforms should I test this on?

Reply:
Not sure, not even sure we need somethgin else than linux, we aren't weirdos after all.

Maybe documentation is enough? Also, from looking at the docu, shutil should at least work relatively well with several platforms, so....